lib/rubocop/cop/betterment/no_strict_loading.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Betterment
+      module NoStrictLoading
+        # This cop ensures that `self.strict_loading_by_default = <any value>` is not set in ActiveRecord models.
+        class ByDefaultForModels < Base
+          extend AutoCorrector
+
+          MSG = 'Do not set `self.strict_loading_by_default` in ActiveRecord models.'
+
+          # @!method strict_loading_by_default_set?(node)
+          def_node_matcher :strict_loading_by_default_set?, <<~PATTERN
+            (send self :strict_loading_by_default= _)
+          PATTERN
+
+          def on_send(node)
+            return unless strict_loading_by_default_set?(node)
+
+            source_range = node.source_range
+
+            add_offense(source_range) do |corrector|
+              buffer    = source_range.source_buffer
+              start_pos = source_range.begin_pos - source_range.column
+
+              # Find the end of the line, including trailing comments and contiguous blank lines
+              match   = buffer.source[start_pos..].match(/.*\n+/)
+              end_pos = start_pos + match[0].length
+
+              corrector.remove(Parser::Source::Range.new(buffer, start_pos, end_pos))
+            end
+          end
+        end
+
+        # This cop ensures that `strict_loading: <any value>` is not set in ActiveRecord associations.
+        class ForAssociations < Base
+          extend AutoCorrector
+
+          MSG = 'Do not set `:strict_loading` in ActiveRecord associations.'
+
+          ASSOCIATION_METHODS = %i(belongs_to has_and_belongs_to_many has_many has_one).freeze
+
+          # @!method association_with_strict_loading?(node)
+          def_node_matcher :association_with_strict_loading?, <<~PATTERN
+            (send nil? {#{ASSOCIATION_METHODS.map(&:inspect).join(' ')}} _ (hash <(pair (sym :strict_loading) _) ...>))
+          PATTERN
+
+          def on_send(node) # rubocop:disable Metrics/PerceivedComplexity
+            return unless association_with_strict_loading?(node)
+
+            hash_node = node.last_argument
+            return unless hash_node.hash_type?
+
+            hash_node.pairs.each do |pair|
+              next unless pair.key.sym_type? && pair.key.value.equal?(:strict_loading)
```

Comment:
In your `def_node_matcher`, you should be able to capture the relevant pair with a `$`.

Reply:
__comment:__ I tried various incarnations of this, but I couldn't make that work as reliably as what I have now.